File: uncompress.py
def uncompress(s):
  output = []
  i = 0
  j = 0
  while j < len(s):
    if s[j].isnumeric():
       j += 1
    else:
      output.append(int(s[i:j]) * s[j])
      j += 1
      i = j
      
  return ''.join(output)

assert uncompress("2c3a1t") == 'ccaaat'
assert uncompress("4s2b") == 'ssssbb'
assert uncompress("4s2b") == 'ssssbb'
assert uncompress("3n12e2z") == 'nnneeeeeeeeeeeezz'
assert uncompress("127y") == 'yyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyy'

File: uncompress.cpp
#include <string>
#include <cctype>
#include <iostream>

std::string uncompress(std::string s) {
  std::string output = "";
  int i = 0;
  int j = 0;
  while (j < s.length()) {
    if (std::isdigit(s[j])) {
      ++j;
    }
    else {
      int count = std::stoi(s.substr(i, j - i));
      output.append(std::string(count, s[j]));
      ++j;
      i = j;
    }
  }
  return output;
}

int main() {

    std::cout << (uncompress("2c3a1t") == "ccaaat") << std::endl;
    std::cout << (uncompress("4s2b") == "ssssbb") << std::endl;
    std::cout << (uncompress("2p1o5p") == "ppoppppp") << std::endl;
    std::cout << (uncompress("3n12e2z") == "nnneeeeeeeeeeeezz") << std::endl;
    std::cout << (uncompress("127y") == "yyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyy") << std::endl;

    return 0;
}